libqfieldsync/offline_converter.py

```diff
@@ -346,6 +349,18 @@ def _convert(self, project: QgsProject) -> None:
         QgsProject.instance().write(str(export_project_filename))
         project.writeProject.disconnect(on_original_project_write)
 
+    def remove_empty_tree_groups_from_project(self, group: QgsLayerTreeGroup) -> None:
+        """
+        Recursively removes any empty groups from the given layer tree group.
+        """
+        for child in group.children():
+            if not isinstance(child, QgsLayerTreeGroup):
+                continue
+
+            self.remove_empty_tree_groups_from_project(child)
```

Comment:
```suggestion

	    # remove recursively
            self.remove_empty_groups_from_layer_tree_group(child)

```

Sorry, I overlooked this recursion here. Place make it explicit and add a line before and after just to highlight it in the function body.

Also a new suggestion for the name of the function, sorry for being annoying with renames.

Feel free to force push with all in a single commit.

Reply:
Great!... Thanks for the suggestions!